src/services/refactors/moveToFile.ts

```diff
@@ -1109,4 +1114,45 @@ function isNonVariableTopLevelDeclaration(node: Node): node is NonVariableTopLev
     }
 }
 
+function moveStatementsToTargetFile(changes: textChanges.ChangeTracker, program: Program, statements: readonly Statement[], targetFile: SourceFile, toMove: ToMove) {
+    const checker = program.getTypeChecker();
+    const sourceExports = new Set<TopLevelDeclarationStatement>();
+    for (const node of toMove.all) {
+        if (isTopLevelDeclarationStatement(node) && hasSyntacticModifier(node, ModifierFlags.Export)) {
+            sourceExports.add(node);
+        }
+    }
+
+    let lastReExport: ExportDeclaration | undefined;
+    for (const node of targetFile.statements) {
+        if (isExportDeclaration(node) && node.moduleSpecifier && isStringLiteralLike(node.moduleSpecifier)) {
+            if (getResolvedModule(targetFile, node.moduleSpecifier.text, getModeForUsageLocation(targetFile, node.moduleSpecifier))
+                && node.exportClause && isNamedExports(node.exportClause) && length(node.exportClause.elements)) {
+                const elements = node.exportClause.elements;
+                const updatedElements = filter(elements, elem => {
+                    const symbol = checker.getSymbolAtLocation(elem.name);

```

Comment:
Pretty sure this is just `elem.symbol` since an `ExportSpecifier` is a `Declaration`